### PR TITLE
Fix parse_ali bug and refactoring

### PR DESCRIPTION
**Pull request by ethantkoenig**
My major changes are:
- Fixing the parse_ali bug
- Making variable names more descriptive
- Add specifications to functions
- Remove stale, commented code

### CODE REVIEW

File: load_ali.m [hunk 1]
```diff
@@ -1,105 +1,102 @@
...
-% Open input streams for alignments in various formats.
-% Basic alignment with transition IDs.
-basic_ali = '/tmp/align3_basic_ali.txt';
-cmd = ['gzcat ', alifile, ' > ', basic_ali];
-system(cmd);
-basic_stream = fopen(basic_ali);
+  % Open input streams for alignments in various formats.
+  % Basic alignment with transition IDs.
+  basic_ali = '/tmp/align3_basic_ali.txt';
+  cmd = ['gunzip -c ', alifile, ' > ', basic_ali];
```
Comment (ethantkoenig) [inline load_ali.m:20]: "gzcat" is equivalent to "gunzip -c". Ubuntu does not support "gzcat" but does support "gunzip", so this change should (hopefully) make this command more portable.

File: load_kaldi_wavscp.m [hunk 2]
```diff
@@ -19,13 +19,13 @@
...
     %disp(j);
-    j = j + 1;
+    %j = j + 1;
     line = fgetl(istream);
 end
 fclose(istream);
 
 function [uid,scp] = parse_wavscp(line)
-    k = findstr(' ',line);
+    k = strfind(line,' ');
```
Comment (ethantkoenig) [inline load_kaldi_wavscp.m:28]: MATLAB recommended using the unambiguous "strfind" instead of the ambiguous "findstr".

File: syllcorpus.m [hunk 3]
```diff
@@ -1,206 +1,186 @@
...
-system('which flac');
-
-utterance_data(ui);
- 
-
-% [PH,SU,PHstart,PHend,SUstart,SUend,WRstart,tra] = parse_ali2(uid,Align_pdf,Align_phone,Tra,P,n)
-
-% Set phone and audio data for k'th utterance.
-% Values are for utterance k.
-    function utterance_data(k)
```
Comment (ethantkoenig) [inline syllcorpus.m:58]: Many of the variables set by this function were never used, and after eliminating the void lines the function became smaller. I've inlined inside the below for loop (where utterance_data was originally called).

**Comment by MatsRooth** [comment]
Bug fix, simplifications, and documentation from Ethan.

**Pull request merged**